src/main/java/edu/carroll/cs389application/web/controller/ImageController.java

```diff
@@ -0,0 +1,162 @@
+
+/**
+ * The ImageController class handles all HTTP requests related to images, including uploading, retrieving, and displaying
+ * images. It interacts with the ImageService and UserService to perform the necessary operations, such as saving and
+ * retrieving images from the database.
+ */
+@Controller
+public class ImageController {
+    /**
+     * This is a basic logging platform that we use throughout the application for logging errors,info, and warnings.
+     */
+    private static final Logger log = LoggerFactory.getLogger(ImageServiceImpl.class);
+
+    /**
+     * This instance of the ImageService will handle laoding images and saving images.
+     */
+    private final ImageService imageService;
+
+    /**
+     * This instance of the UserService will help navigate active users and help save images.
+     */
+    private final UserService userService;
+
+    /**
+     * Constructor for the ImageController class.
+     *
+     * @param imageService The ImageService used to perform operations related to images.
+     * @param userService  The UserService used to perform operations related to users.
+     */
+    public ImageController(ImageService imageService, UserService userService) {
+        this.imageService = imageService;
+        this.userService = userService;
+    }
+
+    /**
+     * Handles file upload requests.
+     *
+     * @param fileForm the form containing information about the image file
+     * @param file     the image file itself
+     * @param result   the result of the validation of the image file
+     * @param model    the model used to store data to be displayed by the view
+     * @param session  the user's session
+     * @return the view to display
+     */
+    @PostMapping("/upload")
+    public String handleFileUpload(@ModelAttribute("fileForm") ImageForm fileForm, @RequestParam("imageFile") MultipartFile file, BindingResult result, Model model, HttpSession session) {
+        String username = (String) session.getAttribute("username");
+        passImages(model, username);
+
+        if (result.hasErrors()) {
+            return "togar";
+        }
+
+        if (!imageService.validateFile(file).equals(ErrorCode.VALID_FILE)) {
+            ErrorCode fileError = imageService.validateFile(file);
+            log.debug(fileError.toString());
+            model.addAttribute("fileError", fileError);
+            return "togar";
+        }
+        log.info("Started upload {}", username);
+
+        fileForm.setImageFile(file);
+        imageService.saveImage(fileForm, userService.loginFromUsername(username));
+        return "redirect:/togar";
+
+    }
+
+    /**
+     * Displays the user's image gallery.
+     *
+     * @param model   the model used to store data to be displayed by the view
+     * @param session the user's session
+     * @return the view to display
+     */
+    @GetMapping("/togar")
+    public String imageGallery(Model model, HttpSession session) {
+        if (session == null) {
```

Comment:
These two validation checks should ALSO be done in `handleFileUpload`.  (Spring Security allows you to do this more easily, but without it you must be more verbose.)